src/Workspaces/Core/Portable/Shared/Extensions/IMethodSymbolExtensions.cs

```diff
@@ -168,13 +168,13 @@ public static IMethodSymbol RenameParameters(this IMethodSymbol method, IList<st
             return updatedMethod.RenameParameters(parameterNames);
         }
 
-        public static IMethodSymbol RemoveInaccessibleAttributesAndAttributesOfType(
-            this IMethodSymbol method, ISymbol accessibleWithin, INamedTypeSymbol removeAttributeType,
-            IList<SyntaxNode> statements = null, IList<SyntaxNode> handlesExpressions = null)
+        public static IMethodSymbol RemoveInaccessibleAttributesAndAttributesOfTypes(
+            this IMethodSymbol method, ISymbol accessibleWithin, params INamedTypeSymbol[] removeAttributeTypes)
         {
             Func<AttributeData, bool> shouldRemoveAttribute = a =>
-                a.AttributeClass.Equals(removeAttributeType) || !a.AttributeClass.IsAccessibleWithin(accessibleWithin);
-            return method.RemoveAttributesCore(shouldRemoveAttribute, statements, handlesExpressions);
+                removeAttributeTypes.Any(attr => attr != null && attr.Equals(a.AttributeClass)) || !a.AttributeClass.IsAccessibleWithin(accessibleWithin);
```

Comment:
oh wow.  That's not how i parsed things originally.  I didn't even realized that ```!a.AttributeClass.IsAccessibleWithin(accessibleWithin)``` was outside the lambda!